pkg/client/restclient/request.go

```diff
@@ -811,7 +813,14 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
 		}
 		if err != nil {
-			return err
+			operr, ok := err.(*net.OpError)
+			if ok && operr.Err.Error() == syscall.ECONNRESET.Error() {
```

Comment:
Can we create a `utilnet.IsConnectionReset(err)` to dry this up a bit?

I think in general this makes sense.  You may need to apply it to other places in request.go - not all code goes through this method.